nvtabular/tools/data_gen.py

```diff
@@ -0,0 +1,442 @@
+        self.dtype = dtype
+        self.distro = distro
+
+    def tupel(self):
+        tupel = []
+        for attr, val in self.__dict__.items():
+            tupel.append(val)
+        return tupel
+
+
+class ContCol(Col):
+    def __init__(
+        self,
+        name,
+        dtype,
+        min_val=0,
+        max_val=1,
+        mean=None,
+        std=None,
+        per_nan=None,
+        width=1,
+        distro=None,
+    ):
+        super().__init__(name, dtype, distro)
+        #         self.dtype = dtype
+        self.min_val = min_val
+        self.max_val = max_val
+        self.mean = mean
+        self.std = std
+        self.per_nan = per_nan
+        self.width = width
+
+
+class CatCol(Col):
+    def __init__(
+        self,
+        name,
+        dtype,
+        cardinality,
+        min_entry_size=None,
+        max_entry_size=None,
+        avg_entry_size=None,
+        per_nan=None,
+        multi_min=None,
+        multi_max=None,
+        multi_avg=None,
+        distro=None,
+    ):
+        super().__init__(name, dtype, distro)
+        #         self.dtype = dtype
+        self.cardinality = cardinality
+        self.min_entry_size = min_entry_size
+        self.max_entry_size = max_entry_size
+        self.avg_entry_size = avg_entry_size
+        self.per_nan = None
+        self.multi_min = multi_min
+        self.multi_max = multi_max
+        self.multi_avg = multi_avg
+
+
+class LabelCol(Col):
+    def __init__(self, name, dtype, cardinality, distro=None):
+        super().__init__(name, dtype, distro)
+        #         self.dtype = dtype
+        self.cardinality = cardinality
+
+
+def _get_cols_from_schema(schema, distros=None):
+    """
+    schema = a dictionary comprised of column information,
+             where keys are column names, and the value
+             contains spec info about column.
+
+    Schema example
+
+    conts:
+        col_name:
+            dtype:
+            min_val:
+            max_val:
+            mean:
+            standard deviation:
+            % NaNs:
+    cats:
+        col_name:
+            dtype:
+            cardinality:
+            min_entry_size:
+            max_entry_size:
+            avg_entry_size:
+            % NaNs:
+            multi_min:
+            multi_max:
+            multi_avg:
+
+    labels:
+        col_name:
+            dtype:
+            cardinality:
+            % NaNs:
+    """
+    cols = {}
+    executor = {"conts": ContCol, "cats": CatCol, "labs": LabelCol}
+    for section, vals in schema.items():
+        cols[section] = []
+        for col_name, val in vals.items():
+            v_dict = {"name": col_name}
+            v_dict.update(val)
+            if distros and col_name in distros:
+                dis = distros[col_name]
+                new_distr = distro_types[dis["name"]](**dis["params"])
+                v_dict.update({"distro": new_distr})
+            cols[section].append(executor[section](**v_dict).tupel())
```

Comment:
So - it seems like the only use for ContCol/CatCol/LabelCol classes is to get created from a dictionary - and then get immediately called 'tupel' on and aren't used afterwards.

I think we should either remove the ContCol/CatCol/LabelCol classes - and just store the dictionary here (```cols[section].append(v_dict)```) or we should return the class and not call tupel (```cols[sections].append(executor[section](**v_dict)```).   I think either way will be safer than relying on the dictionary order of the class in create_conts/create_cats/create_cols